testing_lib_common.md

```diff
@@ -0,0 +1,58 @@
+# Testing local changes to lib-common
+
+When developing changes in lib-common, prior to submitting a PR it's handy
+to be able to test those changes with another operator that uses lib-common.
+This topic describes how [run the operator locally](running_local_operator.md),
+with it configured to use a local lib-common repository.
+
+Note that this topic does not cover unit tests in lib-common.
+
+## Prequisites
+
+* You forked the [lib-common](https://github.com/openstack-k8s-operators/lib-common) repository, and have a local clone of your fork
+* You have an operator running locally that you intend to use to test your lib-common changes
+
+## Procedure
+
+1. Modify your operator's go.mod file(s)
+
+Locate and edit your operator's go.mod files (when in doubt, try: `find
+. -name go.mod`). Note the `require` entries for lib-common, and identify the
+one associated lib-common code you're working on. This example will be for
+changes to lib-common's **common** module.
+
+```golang
+require (
+	...
+	github.com/openstack-k8s-operators/lib-common/modules/common v0.1.1-0.20231001084618-12369665b166
+	...
+)
+```
+
+At the bottom of each go.mod file, add a `replace` line that references your
+local lib-common repository. The reference may use an absolute path:
+
+```
+replace github.com/openstack-k8s-operators/lib-common/modules/common => /path/to/your/lib-common/modules/common
```

Comment:
could you please what you have to add in go.mod to be able to build your operator with a custom lib-common. This also allows e.g. to build/test an openstack-operator with a custom service operator

```
replace github.com/openstack-k8s-operators/lib-common/modules/common => github.com/<yourgithub>/lib-common/modules/common <commit hash>
```

running the `make tidy` will then replace the commit hash with the sudo version. after that you can build your operator with the custom lib-common as describe in https://github.com/openstack-k8s-operators/docs/blob/main/image_build.md

Reply:
Good idea, done!